packages/api/apollo-server/types.ts

```diff
@@ -1,13 +1,82 @@
 import { gql } from 'apollo-server-core'
 
 const Types = gql`
+  input UserInput {
+    id: String
+    firstName: String
+    lastName: String
+    utorID: String
+    userType: UserType
+  }
+
   type Query {
-    getUser: User
-    getUsers: [User]
+    getUser(id: String, utorID: String, email: String): User
+    getUsers(users: [UserInput!]!): [User!]!
+    getAssignment(id: String): Assignment
+    getAssignments: [Assignment!]!
+    getSubmission(id: String): Submission
```

Comment:
For any query/mutation. Always return the entity unless there's a very good reason for it to be nullable. There's a couple of these below.